### PR TITLE
Remove errors for nodes without NodeFeatures

### PR DESCRIPTION
**Pull request by Kavinraja-G**
Fix: #1894

### CODE REVIEW

**Comment by marquiz** [comment]
Thank you @Kavinraja-G for the PR. At a quick look this looks like a good fix for the problem. I have to take a more thoughtful look, later.

Have you been able to verify this?
/ok-to-test

**Comment by Kavinraja-G** [comment]
> Have you been able to verify this?

Looks like I need to modify the unit tests as well. It expects errors for missing nodeFeatures

**Comment by marquiz** [comment]
```
pkg/apis/nfd/nodefeaturerule/rule.go:222:4: unstructured logging function "Infof" should not be used
```

File: pkg/apis/nfd/nodefeaturerule/rule_test.go [hunk 1]
```diff
@@ -53,9 +53,6 @@ func TestRule(t *testing.T) {
 	assert.Nilf(t, err, "unexpected error: %v", err)
 	assert.Equal(t, r1.Labels, m.Labels, "empty matcher should have matched empty features")
 
-	_, err = Execute(r2, f)
-	assert.Error(t, err, "matching against a missing feature should have returned an error")
```
[the remark this answers is not in the document]
Comment (Kavinraja-G) [reply pkg/apis/nfd/nodefeaturerule/rule_test.go:57]: That make sense, updated https://github.com/kubernetes-sigs/node-feature-discovery/pull/1988/commits/ec0341a25764a66de7c90de9e506b51f9d0c9c2d

File: pkg/apis/nfd/nodefeaturerule/rule_test.go [hunk 2]
```diff
@@ -54,7 +54,7 @@ func TestRule(t *testing.T) {
 	assert.Equal(t, r1.Labels, m.Labels, "empty matcher should have matched empty features")
 
 	_, err = Execute(r2, f)
-	assert.Error(t, err, "matching against a missing feature should have returned an error")
+	assert.NoError(t, err, "matching against a missing feature should not have returned an error")
```
Comment (marquiz) [inline pkg/apis/nfd/nodefeaturerule/rule_test.go:57]: We should add similar asserts to the result as we did for the other test case below. Also, could you squash the commits into one?

Comment (Kavinraja-G) [reply pkg/apis/nfd/nodefeaturerule/rule_test.go:57]: Done!

**Review by marquiz — APPROVED** [review]
Thank you @Kavinraja-G for the contribution. This will make logs cleaner and should make users less confused.

**Comment by marquiz** [comment]
@Kavinraja-G could you rebase this?

**Review by marquiz — COMMENTED** [review]
/lgtm

**Pull request merged**